### PR TITLE
Update onboarding for ab testing android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/42170

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by hffvld** [comment]
Verified on `Galaxy Tab S8` using version(s):
```
Device/OS: Galaxy Tab S8 / gts8wifixx-user 14 UP1A.231005.007 release-keys
Brave build: 1.75.21
Chromium: 131.0.6778.69 (Official Build) canary (64-bit)
```
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-browser/issues/42170#issue-2642209610
2. *Verify*

**ACTUAL RESULTS:**
- Verified that with experiment `A` (new onboarding) `adb shell "echo '--enable-features=BraveDayZeroExperiment:variant/a' > /data/local/tmp/brave-test-day-zero-expt"` enabled when going through Brave onboarding:
  - `Brave animation` is shown.
  - System/OS level `default browser` modal is shown.
  - `p3a` modal is shown.
  - `NTP` is shown.
  - Verified that closing and relaunching Brave during onboarding shows:
    - `Brave animation` and System/OS  `default browser` modal.
    - `Brave animation`, System/OS  `default browser` and `p3a` modal (if tapped `Cancel` on `default browser` modal).
    - `Brave animation` and `p3a` modal.
- Verified that with experiment `B` (default onboarding) `adb shell "echo '--enable-features=BraveDayZeroExperiment:variant/b' > /data/local/tmp/brave-test-day-zero-expt"` enabled when going through Brave onboarding:
  - `Brave animation` is shown.
  - `Notification` modal is shown.
  - Brave's level `default browser` modal is shown.
  - `p3a` modal is shown.
  - `NTP` is shown.
  - Verified that closing and relaunching Brave during onboarding shows:
     - `Brave animation` and `Notification` modal.
     - `Brave animation` and Brave's level `default browser` modal.
     - `Brave animation`, `default browser` (without an option to choose the browser) and `p3a` modals.

---------------------------

`New onboarding` with experiment `A` enabled

https://github.com/user-attachments/assets/34a76af9-4b0f-4838-9d3f-f3affb63d345

`Old onboarding` with experiment `B` enabled

https://github.com/user-attachments/assets/29efeb08-97b1-4904-a7c3-0790b38fad24

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 15` using the following build(s):

```
Brave | 1.75.21 Chromium: 131.0.6778.69 (Official Build) canary (64-bit)
--- | ---
Revision | 77b46ded5480ad3885206464169d6969dfc65abd
OS | Android 15; Build/AP41.240925.009; 35; REL
```

### `Test Case #1`

* connected the `Pixel 6` to the `Ubuntu 24.04 x64` VM and ran the following command via the terminal
  * `adb shell "echo '--enable-features=BraveDayZeroExperiment:variant/a' > /data/local/tmp/brave-test-day-zero-expt"`

image

* ensured that the `Brave` animation is being displayed (same as the previous onboarding)
* ensured that the `Notification` permission isn't being displayed at the beginning of onboarding
* launched `Brave` and ensured that `Set default browser` system dialog is displayed right after Brave has launched
  * ensured that selecting `Brave` as the default worked without any issues
  * ensured that selecting other browsers other than `Brave` as the default worked without any issues
  * ensured that selecting `Cancel` worked without any issues and left the current browser as the default
  * ensured that restarting `Brave` while the `default browser` system dialog is opened doesn't cause any issues
  * ensured that clicking on `Cancel` and then restarting Brave while on the `P3A` screen doesn't cause any issues
* ensured that the `Diagnostic` & `P3A` modals work as before (this hasn't been modified/touched)
* ensured that the NTP is being loaded without any issues/failures once the onboarding as been completed
* ensured that the `Brave News` card is visible via `NTP`
* ensured that `SNTP` images are being displayed without any issues
* ensured that `Brave Rewards` is visible via the omnibox

https://github.com/user-attachments/assets/0aff32d9-6404-4c54-833a-0056c0d3ffb5

**`Example`** | **`Example`** | **`Example`**
---------------|-----------------|---------------
 |  | 

### `Test Case #2`

* connected the `Pixel 6` to the `Ubuntu 24.04 x64` VM and ran the following command via the terminal
  * `adb shell "echo '--enable-features=BraveDayZeroExperiment:variant/b' > /data/local/tmp/brave-test-day-zero-expt"`

image

Basically ensured that the same onboarding as we're currently using is being displayed/used as per the following:

https://github.com/user-attachments/assets/07dd62ff-5cee-426e-9d5e-d7bc68c40148
